### ISSUE TITLE
EventRender still renders events if return false

### ISSUE THREAD

**Issue by Troymlandon** [fullcalendar/fullcalendar-angular#228]
#95 Hopefully I'm doing this right..   I'm finding that even if the function tied to **eventRender** returns false, the event still shows on the calendar.

I've edited the basic Angular example on Stackblitz to show this with a function called **doNotRenderEvents**.

https://stackblitz.com/edit/github-vbbbrp

**Comment by jadkik** [comment]
I ran into the same issue when trying to change the rendered element (i.e. a `div` instead of an `a` tag) and it looks like this is a known issue when I looked through the code.

In [this file](https://github.com/fullcalendar/fullcalendar-angular/blob/163caa6e1b95d459bd8442a68b97b544dbb526bf/projects/fullcalendar/src/lib/fullcalendar-options.ts), you can see:

```
  // TODO: make these inputs...
  'eventRender',
```

And in [this one](https://github.com/fullcalendar/fullcalendar-angular/blob/6b04d74d3a56bbb3c70a1b0eeeeb6b5327f5aaf5/projects/fullcalendar/src/lib/fullcalendar.component.ts):

```
    OUTPUT_NAMES.forEach(outputName => {
      options[outputName] = (...args) => {
        this[outputName].emit(...args);
      };
    });

    // do after outputs, so that inputs with same name override
INPUT_NAMES.forEach(inputName => {
```
Two problems:

  1. `eventRender` is not an input name (so you can't pass the function as an argument)
  2. There is no return value for `this[outputName].emit(...args);`. I think since the event emitter cannot read a return value, they want to make it an input of type function?

I found a workaround here:

https://github-vbbbrp-ul6wuk.stackblitz.io

Basically you override the option after they set it using (in the component where you're using it):

```
  @ViewChild('calendar') calendarComponent: FullCalendarComponent;

  ngAfterViewInit() {
    this.calendarComponent.getApi().setOption('eventRender', (info) => {
      return this.customEventRenderWithReturnValue(info);
    });
  }
```

Hope this helps until it's properly solved.

**Comment by iturn** [comment]
All works well like this;
```
<full-calendar
  #calendar
></full-calendar>
```


```
@ViewChild("calendar", { static: true }) calendarRef: FullCalendarComponent;

  constructor(private renderer: Renderer2) {}

  ngAfterViewInit() {
    this.calendarRef.getApi().setOption("eventRender", info => {
      return this.onRenderEvent(info);
    });
  }

  public onRenderEvent(info) {
    const eventElRef: ElementRef = info.el;

    this.renderer.setStyle(eventElRef, "color", "blue");
  }
```

**Comment by acerix** [comment]
Sorry for the delay in responding to your issue, I have just begun triaging on this repo.

It sounds like this is an actual bug in v4, but `eventRender` no longer exists in v5 and hiding events with `setProp('display', 'none')` would not have this issue.

https://fullcalendar.io/docs/upgrading-from-v4#event-rendering
